### PR TITLE
Updates to documentation

### PR DESCRIPTION
**Pull request by helenb**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1344513237)

### Description of Changes Made

This MR updates the documentation with information from the tech spec for general project principles, adds links to external docs, and adds FE details on streamfield blocks and sustainability. It also adds items to the documentation index, covers a few formatting tweaks and update the readme.

### How to Test

Run `mkdcos serve` when ssh'd into the web container and view the docs at http://0.0.0.0:8001/

### Screenshots

<details>
  <summary>Expand to see more</summary>

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [x] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [ ] Not required

#### Browser testing

- [ ] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [x] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] New font variants have not been added
- [x] Not required

### CODE REVIEW

File: docs/front-end/streamfield-blocks.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Streamfield blocks
+
+We want our streamfield blocks to be as flexible as possible and re-usable on different page types. Therefore when styling the blocks please apply the following principles:
+
+- Ensure consistent spacing between blocks
```
Comment (chris-lawton) [inline docs/front-end/streamfield-blocks.md:5]: Suggestion: could mention using the `$spacer` vars here https://github.com/torchbox/torchbox.com/blob/integration/sprint-1/tbx/static_src/sass/config/_variables.scss#L177 (not yet in `main`)

File: docs/front-end/sustainability.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Sustainability
+
+In order to make the site more sustainable we ensure that images are served in a suitable size for the screen they are displayed on, whether they are background images or images loaded with an <img> tag. We can use <picture> elements with srcset and sizes. See https://docs.wagtail.org/en/v5.2.2/advanced_topics/images/image_file_formats.html#using-the-picture-element.
```
Comment (chris-lawton) [inline docs/front-end/sustainability.md:3]: Suggestion: Is it worth mentioning that images should be in `webp` format?

**Comment by chris-lawton** [comment]
Thanks @helenb - looks great!

**Review by chris-lawton — APPROVED** [review]

File: docs/front-end/sustainability.md (hunk 2 above)
Comment (helenb) [reply docs/front-end/sustainability.md:3]: good catch - I have added that in.

**Pull request merged**